src/vulkan/graphics_pipeline.cc

```diff
@@ -582,8 +925,16 @@ Result GraphicsPipeline::Draw(const DrawArraysCommand* command,
   if (!r.IsSuccess())
     return r;
 
+  if (pipeline_ != VK_NULL_HANDLE &&
+      command->IsPipelineDataDifferentFromPreviousOne()) {
+    r = ResetPipeline();
```

Comment:
Any reason to just not always do this?

Reply:
I think creating new graphics pipeline for every draw commands would cost a lot, because graphics pipeline is involved with my setups.
If we do not have different PipelineData, we do not need to create new one.

Do you think it would be better to create new graphics pipeline for every draw commands regardless of performance at this moment?